### DISCUSSION TITLE
Status project

### DISCUSSION THREAD

**Discussion by louisvangeldrop** [microsoft/devicescript#464] [General]
Just out of curiosity, what is the status of this super good project? Is it a research-project or will it have a longer continuity?

**Comment by pelikhan** [comment]
DeviceScript is still in the incubation/experimentation phase. We think that a large factor of long term success will be community engagement, specially around adding driver support. 
This is why it's critical for us to enable setWatch/clearWatch to enable users to write drivers.

**Reply by louisvangeldrop** [reply]
Thanks for your update about the status of the project. Indeed programming drivers in Typescript could certainly increase the community engagement. There are many Typescript programmers and there are a lot of C/C++ programmers, but there not that many programmers that program in both languages.

**Reply by louisvangeldrop** [reply]
Please also have a look at the digitalPulse function

**Reply by pelikhan** [reply]
https://www.espruino.com/Reference#l__global_digitalPulse

**Reply by louisvangeldrop** [reply]
I hope that you have enjoyed your holidays :-). Is there already any progress with the setWatch/digitalPulse functions implementations?

**Reply by pelikhan** [reply]
not yet @mmoskal fyi

**Comment by louisvangeldrop** [comment]
After a long time of silence on this project, I have to ask again: "What is the status of this super good project?"

**Reply by pelikhan** [reply]
The status has not changed much. The project is still in an incubation phase and we take contributions here and there, and we fix bugs reported by our users. However, we are not actively adding new features at the moment.

**Reply by louisvangeldrop** [reply]
That is a pity, since I was hoping for the setWatch and digitalPulse functions. Would make it a lot easier to implement drivers.

**Reply by pelikhan** [reply]
@mmoskal could we consider having something like raspberry pio to make it easier to run real time portions of the program?
